coderadar-test/src/test/java/io/reflectoring/coderadar/rest/query/GetCommitLogControllerTest.java

```diff
@@ -0,0 +1,81 @@
+import io.reflectoring.coderadar.query.domain.CommitLog;
+import io.reflectoring.coderadar.rest.ControllerTestTemplate;
+import io.reflectoring.coderadar.rest.domain.IdResponse;
+import java.net.URL;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MvcResult;
+
+public class GetCommitLogControllerTest extends ControllerTestTemplate {
+
+  private Long projectId;
+
+  @BeforeEach
+  void setUp() throws Exception {
+    URL testRepoURL = this.getClass().getClassLoader().getResource("test-repository");
+    CreateProjectCommand command1 =
+        new CreateProjectCommand(
+            "test-project", "username", "password", testRepoURL.toString(), false, null, null);
+    MvcResult result =
+        mvc()
+            .perform(
+                post("/projects").contentType(MediaType.APPLICATION_JSON).content(toJson(command1)))
+            .andReturn();
+
+    projectId = fromJson(result.getResponse().getContentAsString(), IdResponse.class).getId();
+  }
+
+  @Test
+  void returnsAllCommitsInProject() throws Exception {
+    MvcResult result =
+        mvc()
+            .perform(
+                get("/projects/" + projectId + "/commitLog")
+                    .contentType(MediaType.APPLICATION_JSON))
+            .andDo(
+                document(
+                    "commit/log",
+                    responseFields(
+                        fieldWithPath("[]")
+                            .description(
+                                "Array of the commits in this project in a git log similar format."),
+                        fieldWithPath("[].refs")
+                            .description("Array of strings with the refs pointing to the commit"),
+                        fieldWithPath("[].parents")
+                            .description("Array of strings with the parents of the commit"),
+                        fieldWithPath("[].hash").description("The hash of the commit."),
+                        fieldWithPath("[].author.name").description("The name of the author"),
+                        fieldWithPath("[].author.email").description("The email of the author"),
+                        fieldWithPath("[].author.timestamp")
+                            .description("The timestamp of the commit"),
+                        fieldWithPath("[].author")
+                            .description("Object containing author information about the commit."),
+                        fieldWithPath("[].subject")
+                            .description(
+                                "The comment of this commit truncated to the first 100 characters."),
+                        fieldWithPath("[].analyzed")
+                            .description("Whether this commit is already analyzed or not."))))
```

Comment:
You document the different response fields here, but you don't include them in the `restapi.adoc` file. We could also document the fields in the same order as we get the response from the server, i.e. swap `parents` and `hash`, and document `author` before documenting the specific fields of `author`.

Reply:
Done